src/main/scala/flow.scala

```diff
@@ -103,11 +131,15 @@ sealed abstract class Comprehension[M[_]]{ // FIXME: what happens if calling a m
   //def apply[T](scope: => T): M[T]
 }
 final class MonadContext[M[_]]{
+  /** transform the surrounding monad at this point */
+  @compileTimeOnly("The MonadContext type only makes sense in a flow comprehension scope and is supposed to be removed by the macro.")
+  def !(transform: M[FlowContext] => M[FlowContext]) = ???
+  /** extract a value from a given Monad */
   @compileTimeOnly("The MonadContext type only makes sense in a flow comprehension scope and is supposed to be removed by the macro.")
-  def apply(transform: M[FlowContext] => M[FlowContext]) = ???
+  def ?[T](monad: M[T]): T = ???
 }
```

Comment:
Can we get by without `!` by having the context be typed as an object of the monad type and define `?` as an implicit extension?